admin-dev/themes/new-theme/js/pages/product/components/categories/categories-manager.ts

```diff
@@ -36,11 +36,11 @@ export default class CategoriesManager {
 
   categoryTreeSelector: CategoryTreeSelector;
 
-  categoriesContainer: HTMLElement;
+  categoriesContainer: HTMLElement | null;
 
-  defaultCategoryInput: HTMLInputElement;
+  defaultCategoryInput: HTMLInputElement | null | undefined;
 
-  addCategoriesBtn: HTMLElement;
+  addCategoriesBtn: HTMLElement | null | undefined;
```

Comment:
can't we set them as optional instead?

Reply:
actually I think this btn shouldn't be optional. Its either HTMLElement or it is an error. (If im being correct, this button must always exist in front end, so why would it be optional?)

Reply:
If you want to shortcut the types, then you can use the non-null assertion operator like `addCategoriesBtn!`, but a dom object can always be null or undefined, as you have no control over the dom generation, this is why `querySelector` returns `Element | undefined`

Reply:
as you said, if the element isn't selectable, it would be undefined and the error would be handled by the JS, it would be 100% error-proof